### PR TITLE
Fix API Gateway test failure if more that 25 APIs in API Gateway

### PR DESCRIPTION
**Pull request by ckthiessen**
This is addressing issue #504. Increased the number of APIs returned from the default 25 to the maximum of 500 and added pagination in case more than 500 APIs exist. All API Gateway tests pass.
I am creating a new PR because I moved my changes onto a different branch to follow contributing conventions.

### CODE REVIEW

File: lib/awspec/helper/finder/apigateway.rb [hunk 1]
```diff
@@ -2,17 +2,23 @@ module Awspec::Helper
   module Finder
     module Apigateway
       def find_apigateway_by_id(id)
-        rest_apis = apigateway_client.get_rest_apis
-        rest_apis.items.each do |item|
-          return item if item.id == id
+        apis = []
+        apigateway_client.get_rest_apis(limit: 500).each do |response|
```
Comment (k1LoW) [inline lib/awspec/helper/finder/apigateway.rb]: Thank you for your commit !!!

Could you use `position:` for paging ? ( see https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/APIGateway/Client.html#get_rest_apis-instance_method )

Comment (ckthiessen) [reply lib/awspec/helper/finder/apigateway.rb]: Hey k1LoW, thank you for reviewing my PR! AWS actually made responses enumerable so this is using the `position:` param even though it doesn't say it explicitly.  I am iterating through the `position`s.  This is the standard for using AWS pagination as defined in the documentation here:  https://docs.aws.amazon.com/sdk-for-ruby/v3/developer-guide/paging-responses.html

Comment (k1LoW) [reply lib/awspec/helper/finder/apigateway.rb]: Oops! You are right!

**Comment by k1LoW** [comment]
Released as v1.18.5.

**Pull request merged**